lib/liveman/survey/surveys.ex

```diff
@@ -8,19 +8,56 @@ defmodule Liveman.Survey.Surveys do
   def list_surveys do
     survey_list_json = fecth_surveys_from_file!()
 
-    Enum.map(survey_list_json, fn survey_json ->
-      build_survey(survey_json)
-    end)
+    get_survey_list(survey_list_json)
   end
 
   defp fecth_surveys_from_file! do
     {_, body} = File.read(@json_file)
     Jason.decode!(body)
   end
 
+  def create_answers(params) do
+    survey = get_survey(params["survey_id"])
+
+    validate(survey, params["questions"])
+  end
+
+  defp validate(nil, _questions), do: {:error, :invalid_survey}
+  defp validate(_survey, nil), do: {:error, :invalid_questions}
+
+  defp validate(survey, questions) do
+    survey_questions_ids = get_survey_question_ids(survey.questions)
+    question_ids = get_question_ids(questions)
+
+    # Check if all submitted questions are belong to the survey
+    if question_ids -- survey_questions_ids == [] do
+      :ok
+    else
+      {:error, :invalid_questions}
+    end
+  end
+
+  defp get_survey_list(survey_list_json) do
+    Enum.map(survey_list_json, fn survey_json ->
+      build_survey(survey_json)
+    end)
```

Comment:
There is another shorter version 🤩  You can checkout [Elixir Capture Operator](https://dockyard.com/blog/2016/08/05/understand-capture-operator-in-elixir) 👍 

```elixir
Enum.map(survey_list_json, &build_survey/1)
```